### DISCUSSION TITLE
v1.0.9 but two different file sizes of 32-bit file?

### DISCUSSION THREAD

**Discussion by mushu999** [saucecontrol/BrotliIIS#10] [Q&A] [answered]
I have the 32-bit v1.0.9 brotli.dll in the \windows\system32\inetsrv directory on our server from right after it was released. I updated from v1.0.5 just fine. When right-click on the DLL it shows the following details:
![image](https://user-images.githubusercontent.com/11066472/149211331-2de7b2d2-b588-4f5c-bae8-6a28e5610a03.png)

But when I download the current v1.0.9 brotli.dll 32-bit version, it has different info in the details of the 32-bit file:
![image](https://user-images.githubusercontent.com/11066472/149211835-ede5bbaa-929e-4af1-a674-c40131be11d9.png)
(note I renamed that DLL with -NEW because it crashes my server)

Wondering why the file sizes are different yet the versions are the same? Also, why does the newest larger version crash my server with error 500 but the original works just fine?

**Comment by saucecontrol** [comment, marked as answer]
The larger file is the 64-bit binary.  `\windows\system32\inetsrv` is the correct location for the 64-bit dll.  The 32-bit dll (if needed) would go into `\windows\syswow64\inetsrv` (sounds backwards, I know, but that's correct 😄).  If you mix up the files in the folders, a load failure would be expected.

**Comment by mushu999** [comment]
Well, except that doesn't match our server setup right now.

![image](https://user-images.githubusercontent.com/11066472/149375334-4b56c79e-458d-4fa2-b952-c1f45c666f66.png)

The image above shows the DLL file in our "system32\inetsrv" directory (the 32-bit location) but note the filesize. According to your answer above it is the 64-bit version...but that wouldn't work if it were the case!

Also unanswered is the fact that in my original issues post (which was converted to this discussion thread) I asked why the filesize is different (by a few bytes) from the current version however the Version in the file details shows the same version value of 1.0.9??

I literally just downloaded the v1.0.9 marked "Latest" and unpacked the file in the 32-bit "x32" folder of the archive (the smaller file) and put it in the directory in my screenshot above and renamed the existing Brotli.DLL file, and that is what caused the crash. So apparently we must be running the 64-bit Brotli.DLL file in the 32-bit directory? Note also that there is NO Brotli.DLL file in the SysWow32 directory path at all.

These weirdnesses worry me, thus why i am asking for clarification. Thank you.

**Reply by saucecontrol** [reply]
You've got that backwards.  `system32` is the folder for 64-bit binaries, so you should not be using the 32-bit dll there. If you need the 32-bit dll (only if you have app pools configured as 32-bit), it goes in `syswow64`.

**Reply by saucecontrol** [reply]
Oh, and I'm not sure why your file sizes are showing different, but I can assure you the binaries have not changed since they were uploaded initially. Could be rounding differently in different places, so you may be better off getting the exact file size from cmd prompt.

**Reply by mushu999** [reply]
Oh crap! Well, I feel stupid. Okay, thank you for the clarification. And now that I read your response I realized it was correct. Too many things in my head, too many fires to put out I guess. That makes sense.

Ugh...

**Reply by saucecontrol** [reply]
Ha, it happens. Those names are a bit of a mess 😄
